processor/src/decode.rs

```diff
@@ -18,74 +18,86 @@ pub const OPCODE_FLAG_FIELD_BIT_POSITION: u64 = IMM_FLAG_FIELD_BIT_POSITION+IMM_
 pub const REG_FIELD_BITS_MASK: u32 = 0xf;
 pub const IMM_FLAG_FIELD_BITS_MASK: u32 = 0x1;
 
-pub fn decode_raw_instruction(raw_inst_str: &str, imm_str: &str) -> Result<(String, u64), ParseIntError>{
+fn parse_hex_str(hex_str: &str) ->  Result<u32, ProcessorError> {
+    let res = u32::from_str_radix(hex_str, 16);
+    if let Err(e) = res {
+        return Err(ProcessorError::ParseIntError);
+    } else {
+        return Ok(res.unwrap());
+    }
+}
+
+pub fn decode_raw_instruction(raw_inst_str: &str, imm_str: &str) -> Result<(String, u64), ProcessorError> {
```

Comment:
This function may return Result<Instruction, ProcessorError>, and all kind of instructions impl to_string.